### DISCUSSION TITLE
Is it possible to set custom header for modals opened by modals manager

### DISCUSSION THREAD

**Discussion by BOHDANff** [mantinedev/mantine#5534] [Q&A]
Hello, while I have been using modals manager there arose the question related to customizing of modal header.

There is the next use case: I need to open the modal with header which has not only the title to the left and close button to the right but also such header should include some button which should be located on the left to the close button. But I have not idea how to do this because `modals.open()` method don't support the parameters for setting custom header. It seems that header of modals which could be opened by modals manager is hard coded and it is not possible to set full header as `React.Node` but only customize its title or/and close button. 

So is there any way to set custom header for modals opened by modals manager?

**Comment by rtivital** [comment]
You can pass any props to the Modal component with the object that you provide in `modals.open()`.

**Reply by BOHDANff** [reply]
I understand, but if there are any fields in the object provided in `modals.open()` which allowed to set custom header?

**Reply by rtivital** [reply]
Yes, set `withCloseButton: false`, render your content in `children`

**Reply by BOHDANff** [reply]
Content in children will be rendered in ModalBody. Is it wright way to render custom header in ModalBody but not in ModalContent?

**Reply by rtivital** [reply]
If you really need to render content in the header, use `title` prop. Otherwise, body is ok.

**Reply by BOHDANff** [reply]
Okay, but ModalBody has some css stylization, for example additional paddings. If I want to render custom header in ModalBody I need to rewrite paddings of ModalBody to zero to allow header to have full modal width. Is it right way? Doesn't it look like crutch?
